packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.test.tsx

```diff
@@ -81,4 +82,67 @@ describe('useFluentProvider_unstable', () => {
       }
     `);
   });
+
+  it('should merge customStyles', () => {
+    const customStylesA: FluentProviderCustomStyleHooks = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useButtonStyles_unstable_A';
+      },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useCompoundButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useCompoundButtonStyles_unstable_A';
+      },
+    };
+
+    const customStylesB: FluentProviderCustomStyleHooks = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useCompoundButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useCompoundButtonStyles_unstable_B';
+      },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      useSplitButtonStyles_unstable: (state: unknown) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (state as any).testResult = 'useSplitButtonStyles_unstable_B';
+      },
+    };
+
+    const DefaultWrapper: React.FC = ({ children }) => <FluentProvider>{children}</FluentProvider>;
+
+    const { result: defaultResult } = renderHook(
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      () => useFluentProvider_unstable({}, React.createRef()),
+      {
+        wrapper: DefaultWrapper,
+      },
+    );
+
+    const Wrapper: React.FC = ({ children }) => (
+      <FluentProvider customStyleHooks_unstable={customStylesA}>{children}</FluentProvider>
+    );
+
+    const { result } = renderHook(
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      () => useFluentProvider_unstable({ customStyleHooks_unstable: customStylesB }, React.createRef()),
+      {
+        wrapper: Wrapper,
+      },
+    );
```

Comment:
Let's split this to two separate `it` statements as TBH I am lost in assertions below.